labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +53,15 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case str: String => s"A string with length ${str.length}"
+      case n: Integer if n > 0 => "A positive integer"
+      case person: Person => s"A person with name: ${person.name}"
+      case seq: Seq[_] if seq.size > 10 => "Seq with more than 10 elements"
+      case v1 :: v2 :: tail => s"first: $v1, second: $v2, rest: $tail"
```

Comment:
This will trigger even if `in` has only two elements; `tail` can be `Nil`. Also, not all `Seq`s are `List`s, and the ones that aren’t don’t match on `::`.